internal/http/processor.go

```diff
@@ -43,8 +46,9 @@ type entryTime struct {
 }
 
 var (
-	clientMap  sync.Map
-	torrentMap sync.Map
+	clientIndex int
+	clientMap   sync.Map
+	torrentMap  sync.Map
 )
```

Comment:
Having this as global variables might not be what you actually want, at least not `clientIndex`.

When the request comes in you instantiate a new processor here: https://github.com/nuxencs/seasonpackarr/blob/07668f10e9ecc1d93958c57428549036504abdd4/internal/http/webhook.go#L30

The processor will only live during the request but the global values will stay. And if you do two requests at the same time it will override the `clientIndex`.

`ProcessSeasonPack` is the method that runs all the sub methods so you could make `getClient()` and `getAllTorrents()` to accept a `clientIndex` as argument like

```golang
func (p processor) getClient(clientIndex int) error {}
```

And remove the global `clientIndex` and assign it `:=` on: https://github.com/nuxencs/seasonpackarr/blob/07668f10e9ecc1d93958c57428549036504abdd4/internal/http/processor.go#L153
That would make it safe for multi clients and multiple requests.

Similar with `torrentMap`. If you have multiple clients you'll add all torrents to that map. Not sure if that is expected or not. If it isn't then I'd move that to live on the `processor` struct instead, so that way it only lives for the duration of the request.

Reply:
That makes a lot of sense, thank you for pointing that out. 👍🏼 
Having the `torrentMap` defined globally speeds up consecutive requests considerably, but it probably makes sense to have it sit in the `processor` struct.